PWGCF/FemtoDream/TableProducer/femtoDreamProducerTask.cxx

```diff
@@ -243,6 +271,27 @@ struct femtoDreamProducerTask {
     ccdb->setCreatedNotAfter(now);
   }
 
+  // Function to calculate invariant mass of resonance
+  float CalculateInvMass(const ROOT::Math::PtEtaPhiMVector part1,
+                         const ROOT::Math::PtEtaPhiMVector part2, float massD1, float massD2)
+  {
+    Double_t invMass = 0;
+
+    float EDaug1 = TMath::Sqrt(
+      massD1 * massD1 + part1.Px() * part1.Px() + part1.Py() * part1.Py() + part1.Pz() * part1.Pz());
+    float EDaug2 = TMath::Sqrt(
+      massD2 * massD2 + part2.Px() * part2.Px() + part2.Py() * part2.Py() + part2.Pz() * part2.Pz());
+
+    float energysum = EDaug1 + EDaug2;
+    float pSum2 = (part2.Px() + part1.Px()) * (part2.Px() + part1.Px()) +
+
+                  (part2.Py() + part1.Py()) * (part2.Py() + part1.Py()) +
+
+                  (part2.Pz() + part1.Pz()) * (part2.Pz() + part1.Pz());
+    invMass = TMath::Sqrt(energysum * energysum - pSum2);
+    return invMass;
+  }
```

Comment:
If the ROOT four vectors are properly initialized, which it seems they are, why it is needed to additionally pass the masses? 
Additionally, if the ROOT four vectors are used shouldn't the ROOT four vector arithmetic be utilized in here to simplify things?